### PR TITLE
feat(playground): add prettier IR output

### LINKED ISSUE #1338: Add prettier output and IR to playground
Instructions: 
* Add prettier to wasm https://github.com/oxc-project/oxc/tree/main/crates/oxc_wasm
* Follow the code structure and export prettier output and IR
* Add more tabs to the playground https://github.com/oxc-project/oxc/tree/main/website
* Somehow glue everything together 😅

### PR DESCRIPTION
**Pull request by shannonrothe**
image

- Close: #1338

### CODE REVIEW

File: crates/oxc_prettier/src/printer/mod.rs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 //! References:
 //! * <https://github.com/prettier/prettier/blob/main/src/document/printer.js>
 
-mod command;
+pub(crate) mod command;
```
Comment (shannonrothe) [inline crates/oxc_prettier/src/printer/mod.rs:6]: Not sure I should be exposing this, open to suggestions!

Comment (shannonrothe) [reply crates/oxc_prettier/src/printer/mod.rs:6]: Actually I think we can just serialize `Doc`

File: crates/oxc_prettier/Cargo.toml [hunk 2]
```diff
@@ -20,6 +20,8 @@ oxc_allocator = { workspace = true }
 oxc_ast       = { workspace = true }
 oxc_syntax    = { workspace = true }
 oxc_span      = { workspace = true }
+serde         = { workspace = true }
+serde_json    = { workspace = true }
```
Comment (Boshen) [inline crates/oxc_prettier/Cargo.toml:24]: Not required by this crate

File: crates/oxc_prettier/src/lib.rs [hunk 3]
```diff
@@ -47,4 +48,9 @@ impl<'a> Prettier<'a> {
         let doc = program.format(&mut self);
         Printer::new(doc, self.source_text, self.options).build()
     }
+
+    pub fn cmds(mut self, program: &Program<'a>) -> Vec<Command<'a>> {
```
Comment (Boshen) [inline crates/oxc_prettier/src/lib.rs:52]: I think we just want the `Doc`, commands are not interesting :-/

Comment (shannonrothe) [reply crates/oxc_prettier/src/lib.rs]: https://github.com/oxc-project/oxc/pull/1350/commits/44bf9a293f5e202a539b7d25621d26681eb42de7

File: crates/oxc_prettier/Cargo.toml (hunk 2 above)
Comment (shannonrothe) [reply crates/oxc_prettier/Cargo.toml]: https://github.com/oxc-project/oxc/pull/1350/commits/0e8ff9da0d36d59a8531da7e62bb425743074ded

**Review by camc314 — COMMENTED** [review]
This is awesome

**Comment by magic-akari** [comment]
image

JSON is not idea resutls. Check the Biome playground.

**Comment by shannonrothe** [comment]
@magic-akari check the latest commits, I moved away from `serde` + JSON to plain strings. It's not formatted as nicely as Biome/Prettier playgrounds but we can iterate on it

**Comment by Boshen** [comment]
Let's iterate on this! Thank you @shannonrothe!

**Pull request merged**